tribe-ext-pdf-tickets/index.php

```diff
@@ -0,0 +1,633 @@
+	 *
+	 * @link https://mpdf.github.io/troubleshooting/known-issues.html
+	 */
+	public function init() {
+		if ( version_compare( PHP_VERSION, '5.3.7', '<' ) ) {
+			_doing_it_wrong(
+				$this->get_name(),
+				'This Extension requires PHP 5.3.7 or newer to work. Please contact your website host and inquire about updating PHP.',
+				'1.0.0'
+			);
+
+			return;
+		}
+
+		// Event Tickets
+		add_action( 'event_tickets_attendees_table_row_actions', array( $this, 'pdf_attendee_table_row_actions' ), 0, 2 );
+
+		add_action( 'event_tickets_orders_attendee_contents', array( $this, 'pdf_attendee_table_row_action_contents' ), 10, 2 );
+
+		add_action( 'event_tickets_rsvp_attendee_created', array( $this, 'do_upload_pdf' ), 50, 1 );
+
+		// Event Tickets Plus: WooCommerce
+		add_action( 'event_ticket_woo_attendee_created', array( $this, 'do_upload_pdf' ), 50, 1 );
+
+		// Event Tickets Plus: Easy Digital Downloads
+		add_action( 'event_ticket_edd_attendee_created', array( $this, 'do_upload_pdf' ), 50, 1 );
+
+		// For generating a PDF on the fly
+		add_action( 'template_redirect', array( $this, 'if_pdf_url_404_upload_pdf_then_reload' ) );
+	}
+
+	/**
+	 * The absolute path to the mPDF library directory.
+	 *
+	 * @return string
+	 */
+	private function mpdf_lib_dir() {
+		$path = __DIR__ . '/vendor/mpdf';
+
+		return trailingslashit( $path );
+	}
+
+	/**
+	 * Get the absolute path to the WordPress uploads directory,
+	 * with a trailing slash.
+	 *
+	 * It will return a path to where the WordPress /uploads/ directory is,
+	 * whether it is in the default location or whether a constant has been
+	 * defined or a filter used to specify an alternate location. The path
+	 * it returns will look something like
+	 * /path/to/wordpress/wp-content/uploads/
+	 * regardless of the "Organize my uploads into month- and year-based
+	 * folders" option in wp-admin > Settings > Media.
+	 *
+	 * @return string The uploads directory path.
+	 */
+	protected function uploads_directory_path() {
+		$upload_dir = wp_upload_dir();
+
+		return trailingslashit( $upload_dir['basedir'] );
+	}
+
+	/**
+	 * Get the URL to the WordPress uploads directory, with a trailing slash.
+	 *
+	 * It will return a URL to where the WordPress /uploads/ directory is,
+	 * whether it is in the default location or whether a constant has been
+	 * defined or a filter used to specify an alternate location. The URL
+	 * it returns will look something like
+	 * http://example.com/wp-content/uploads/ regardless of the current
+	 * month we are in.
+	 *
+	 * @return string The uploads directory URL.
+	 */
+	protected function uploads_directory_url() {
+		$upload_dir = wp_upload_dir();
+
+		return trailingslashit( $upload_dir['baseurl'] );
+	}
+
+	/**
+	 * Check if one string (haystack) starts with another string (needle).
+	 *
+	 * @param $full  The haystack string.
+	 * @param $start The needle string.
+	 *
+	 * @return bool
+	 */
+	private function string_starts_with( $full, $start ) {
```

Comment:
Unused method, could remove if you wanted.

Reply:
good catch